src/libmongoc/tests/test-mongoc-topology-reconcile.c

```diff
@@ -140,8 +137,10 @@ _test_topology_reconcile_rs (bool pooled)
     * server0 is selected, server1 is discovered and added to scanner.
     */
    BSON_ASSERT (selects_server (client, secondary_read_prefs, server0));
-   BSON_ASSERT (
-      get_node (client->topology, mock_server_get_host_and_port (server1)));
+   if (!pooled) {
```

Comment:
Some tests reach inside the mongoc_topology_scanner_t to make assertions. This is only applicable for single threaded now.